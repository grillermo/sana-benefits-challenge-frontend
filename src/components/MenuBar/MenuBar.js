import { useContext } from "react";

import { Link } from "wouter";

import AuthContext from '../../contextProviders/authContext'

const SignedInLinks = () => (
  <li>
    <Link href='/sign_out'>
      <a href='/sign_out'> Sign out</a>
    </Link>
  </li>
)

const SignedOutLinks = () => (
  <>
    <li>
      <Link href='/sign_up'>
        <a href='/sign_up'>Sign up</a>
      </Link>
    </li>
    <li>
      <Link href='/sign_in'>
        <a href='/sign_in'>Sign in</a>
      </Link>
    </li>
  </>
)

export default function MenuBar(){
  const { state: authState } = useContext(AuthContext);

  return(
    <nav>
      <ul>
        <li>
          Sana Warnings
        </li>
      </ul>
      <ul>
        {authState.isAuthenticated ? <SignedInLinks /> : <SignedOutLinks />}
      </ul>
    </nav>
  )
}
