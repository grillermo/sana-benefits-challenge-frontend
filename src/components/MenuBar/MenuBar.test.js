import React from "react";
import { mount } from 'enzyme';
import {render, screen, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import MenuBar from './MenuBar'

import AuthContext from '../../contextProviders/authContext'

describe('MenuBar', () => {
  describe('signed in user', () => {
    it('matches the snapshot', async () => {
      const menuBar = mount(
        <AuthContext.Provider
          value={{state: { isAuthenticated: true }}}
        >
          <MenuBar />
        </AuthContext.Provider>
      )

      expect(menuBar).toMatchSnapshot();
    })
  })

  describe('signed out user', () => {
    it('matches the snapshot', async () => {
      const menuBar = mount(
        <AuthContext.Provider
          value={{state: { isAuthenticated: false }}}
        >
          <MenuBar />
        </AuthContext.Provider>
      )

      expect(menuBar).toMatchSnapshot();
    })
  })
})

