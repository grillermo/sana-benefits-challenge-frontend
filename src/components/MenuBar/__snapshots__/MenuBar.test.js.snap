// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar signed in user matches the snapshot 1`] = `
<nav>
  <ul>
    <li>
      Sana Warnings
    </li>
  </ul>
  <ul>
    <li>
      <a
        href="/sign_out"
        onClick={[Function]}
        to={null}
      >
         Sign out
      </a>
    </li>
  </ul>
</nav>
`;

exports[`MenuBar signed out user matches the snapshot 1`] = `
<nav>
  <ul>
    <li>
      Sana Warnings
    </li>
  </ul>
  <ul>
    Array [
      <li>
        <a
          href="/sign_up"
          onClick={[Function]}
          to={null}
        >
          Sign up
        </a>
      </li>,
      <li>
        <a
          href="/sign_in"
          onClick={[Function]}
          to={null}
        >
          Sign in
        </a>
      </li>,
    ]
  </ul>
</nav>
`;
